src/mlpack/tests/hdbscan_test.cpp

```diff
@@ -0,0 +1,101 @@
+/**
+ * @file hdbscan_test.cpp
+ *
+ * Test the HDBSCAN implementation.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/hdbscan/hdbscan.hpp>
+
+
+#include <boost/test/unit_test.hpp>
+#include "test_tools.hpp"
+
+using namespace mlpack;
+using namespace mlpack::hdbscan;
+using namespace mlpack::distribution;
+
+BOOST_AUTO_TEST_SUITE(HDBSCANTest);
+
+BOOST_AUTO_TEST_CASE(singleClusterTest)
+{
+  // Generate 5000 points on 0.1 radius circle.
+  arma::mat points(2, 5000);
+  double pi = 3.14156;
+  double theeta;
+  for (size_t i = 0; i < 5000; i++)
+  {
+    theeta = (double)i/5000.0;
+    points(0, i) = 0.1*sin(2*pi*theeta);
+    points(1, i) = 0.1*cos(2*pi*theeta);
+  }
+
+  HDBSCAN<> h1(5, true);
+  arma::Row<size_t> assignments;
+  h1.Cluster(points, assignments);
+
+  // Some points in the circle will get classified as noise
```

Comment:
This doesn't seem to mesh with the test condition; is the comment correct?

Reply:
Yes, unfortunately it is.

Initially I thought there was an error and added a rounding term eps_error, but then I realized eps_error dpends on distance between points in dataset and is not the correct way to go.

I tried to dig into it. Here is what I found (I don't know if it helps but this is the reason for some points getting classified as noise): When we are creating a MST, though all the points are equivalent, some points will be considered first and will be present with shorter edges in the MST. 

Example : For minClusterSize = 5
Say point p6 has neighbors p1, p2, p3, p4, p5. Once all these 5 points are inserted, when p6 will be considered all edges that are considered with p6 will have distance ( ``` > dCore(p6)```) and p6 will be not equivalent to p1 in the MST, though we know it is.

Reply:
I think we are talking about something different... what I mean is this: the comment says that some points will be classified as noise, but the actual test condition (on line 40) tests that no points are classified as noise.

Unfortunately I don't understand the situation that the rest of your explanation is referring to; can you clarify please?